finetune/model.py

```diff
@@ -216,3 +231,78 @@ def optimizer(lr):
         return tf.estimator.EstimatorSpec(mode=mode, loss=train_loss, eval_metric_ops=metrics)
 
     return _model_fn
+
+
+def get_separate_model_fns(target_model_fn, predict_op, predict_proba_op, build_target_model, encoder, target_dim,
+                 label_encoder, saver, portion, build_attn):
+
+    def _featurizer_model_fn(features, labels, mode, params):
+        assert mode == tf.estimator.ModeKeys.PREDICT, "mode MUST be predict - model fns should not be separated on train"
+        X = features["tokens"]
+        featurizer_state = params.base_model.get_featurizer(
+                X,
+                encoder=encoder,
+                config=params,
+                train=False
+        )
+        predictions = {'features':featurizer_state['features'],
```

Comment:
spacing -- add space after `:`

Might be worth adding a formatter like `black` to your editor just to clean up the little things.

Reply:
https://black.readthedocs.io/en/stable/